### PR TITLE
Protobuf linter

### PR DESCRIPTION
**Pull request by fernandofcampos**
## Purpose of Changes and their Description

Use Buf as a linter for protobufs. 

## Link(s) to Ticket(s) or Issue(s) resolved by this PR

https://linear.app/alloralabs/issue/PROTO-2132

## Are these changes tested and documented?

Covered by existing unit tests and integration tests.

## Still Left Todo

*Fill this out if this is a Draft PR so others can help.*

### CODE REVIEW

**Comment by relyt29** [comment]
I looked over some of the changes but I just want to make explicitly sure: none of the actual contents of the `.proto` files have been changed, right, the protos are just being reformatted with a different formatter right?

**Comment by relyt29** [comment]
oh another question: if we have this new linter, what will be the new formatter command to format our protobufs for us (and does it have a visual studio code integation)

**Comment by fernandofcampos** [comment]
> oh another question: if we have this new linter, what will be the new formatter command to format our protobufs for us (and does it have a visual studio code integation)

The linter and the formatter use the same tool, Buf, that was already being used to generate the protobufs. 

If you run `make` on `x/emissions/Makefile` (for instance) it will format, lint and generate. But there is a separated command to format: `make proto-format`

Let me know if you want me to change it. I will check about VSCode.

**Comment by fernandofcampos** [comment]
> oh another question: if we have this new linter, what will be the new formatter command to format our protobufs for us (and does it have a visual studio code integation)

There is a VSCode extension: https://marketplace.visualstudio.com/items?itemName=bufbuild.vscode-buf

File: x/emissions/Makefile [hunk 1]
```diff
@@ -1,22 +1,8 @@
...
-.DEFAULT_GOAL := build
+.DEFAULT_GOAL := proto-all
 
 DOCKER := $(shell which docker)
 
-#################
-###   Build   ###
-#################
-
-build:
```
Comment (relyt29) [inline x/emissions/Makefile:10]: can we not delete the `build` and `test` commands? I use them

File: x/mint/Makefile [hunk 2]
```diff
@@ -1,26 +1,8 @@
...
-.DEFAULT_GOAL := build
+.DEFAULT_GOAL := proto-all
 
 DOCKER := $(shell which docker)
 
-#################
-###   Build   ###
-#################
-
-build:
```
Comment (relyt29) [inline x/mint/Makefile:10]: same comment here, I object to the removal of `make build` and `make test`

**Review by relyt29 — CHANGES_REQUESTED** [review]
Ok I looked over the commit you highlighted and also then the total PR 

left two comments about a minor thing, please don't delete `make test` and `make build` other than that I'm good with the changes

**Review by relyt29 — APPROVED** [review]

**Pull request merged**